pkg/metering/config.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2025 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metering
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/metering_sdk/storage"
+)
+
+// Config represents the configuration for metering.
+type Config struct {
+	Type    storage.ProviderType `yaml:"type,omitempty" toml:"type,omitempty" json:"type,omitempty" reloadable:"false"`
+	Region  string               `yaml:"region,omitempty" toml:"region,omitempty" json:"region,omitempty" reloadable:"false"`
+	Bucket  string               `yaml:"bucket,omitempty" toml:"bucket,omitempty" json:"bucket,omitempty" reloadable:"false"`
+	Prefix  string               `yaml:"prefix,omitempty" toml:"prefix,omitempty" json:"prefix,omitempty" reloadable:"false"`
+	RoleARN string               `yaml:"role-arn,omitempty" toml:"role-arn,omitempty" json:"role-arn,omitempty" reloadable:"false"`
+}
+
+func (c *Config) adjust() error {
+	if len(c.Type) == 0 {
+		c.Type = storage.ProviderTypeS3
+	}
+	if len(c.Region) == 0 {
+		return errors.New("region is required for the metering config")
+	}
+	if len(c.Bucket) == 0 {
+		return errors.New("bucket is required for the metering config")
+	}
+	if c.Type == storage.ProviderTypeS3 && len(c.RoleARN) == 0 {
```

Comment:
Currently, we only need to support role arn, right? Is the long-term plan to support all defined provider types? I think this could be slightly encapsulated, with authentication information in a separate structure, and each provider also having a separate structure.

And you can comment the supported range

Reply:
Very fair suggestion, but at the moment we haven’t worked out a fully refined configuration recommendation with the SDK provider, so for now we just wrote it in a rough-and-ready way to simply “get it running.”

Reply:
But this should be a public configuration, can we give it more consideration?